### ISSUE TITLE
x/tools/gopls: requests fail with type errors

### ISSUE THREAD

**Issue by 0netrickp0ny** [golang/go#36249]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.13.5 darwin/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/yuki/Library/Caches/go-build"
GOENV="/Users/yuki/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/yuki/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/Cellar/go/1.13.5/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.13.5/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/yuki/Documents/GitHub/go_interpreter/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/sk/tgf7xxtj0534t8wwr4h0g0sc0000gn/T/go-build223359251=/tmp/go-build -gno-record-gcc-switches -fno-common"
</pre></details>

### What did you do?

Run Vscode with the latest  vscode-go extension and gopls language server on.

I opened random go file, gopls logs:
<pre>
[Error - 12:13:36 AM] Request textDocument/codeAction failed.
  Message: /Users/yuki/Documents/GitHub/go_interpreter/hello.go:14:1: expected declaration, found Pri
  Code: 0 
</pre>

And autosuggest is broken too.
<img width="1440" alt="Screen Shot 2019-12-23 at 0 13 49" src="https://user-images.githubusercontent.com/30386540/71323890-91484a00-251b-11ea-93c6-a419907223b0.png">

Similar Case[#31492](https://github.com/golang/go/issues/31492)
I create .code-workspace file, but still does not work.
<pre>
{
  "folders": [
    {
      "path": "/Users/yuki/Documents/GitHub/go_interpreter"
    }
  ],
</pre>

### What did you expect to see?
gopls works with vscode.

### What did you see instead?
Request textDocument/* failed.

**Comment by muirdm** [comment]
Currently you have a syntax error which gopls is reporting: `expected declaration, found Pri`. What do you expect "Pri" to complete to?

Only func/var/type/const declarations are valid at the file level. gopls in general does not work well when faced with syntax errors.

**Comment by 0netrickp0ny** [comment]
> Currently you have a syntax error which gopls is reporting: `expected declaration, found Pri`. What do you expect "Pri" to complete to?
> 
> Only func/var/type/const declarations are valid at the file level. gopls in general does not work well when faced with syntax errors.

Thanks for replying, This is just one example.

gopls doesn't work yet. For example, I have enabled completeUnimported, but the predictions show different ones.
<img width="1440" alt="Screen Shot 2019-12-31 at 9 59 22" src="https://user-images.githubusercontent.com/30386540/71606539-159e6b00-2bb5-11ea-9d41-e9295fbed70d.png">

This is my settings.json.
I enabled the language server.
<img width="1440" alt="Screen Shot 2019-12-31 at 10 00 16" src="https://user-images.githubusercontent.com/30386540/71606568-42528280-2bb5-11ea-9e33-838c5b06f6a1.png">
